storage/kubernetes/storage.go

```diff
@@ -77,15 +79,46 @@ func (c *Config) open(logger logrus.FieldLogger, errOnTPRs bool) (*client, error
 		return nil, err
 	}
 
-	cli, err := newClient(cluster, user, namespace, logger)
+	cli, err := newClient(cluster, user, namespace, logger, c.APIVersion)
 	if err != nil {
 		return nil, fmt.Errorf("create client: %v", err)
 	}
 
 	ctx, cancel := context.WithCancel(context.Background())
 
+	if c.UseCRD {
+		if !cli.createCustomResourceDefinitions() {
```

Comment:
Instead of reproducing this logic, I'd recommend just replacing the `createThirdPartyResources` call with something that conditionally creates either CRDs or TPRs. E.g.

```
// Create either TPRs or CRDs.
cli.registerCustomResoruces(c.UseCRD)
```